src/plan/controller_collector_context.rs

```diff
@@ -50,6 +50,15 @@ impl<VM: VMBinding> ControllerCollectorContext<VM> {
     }
 
     pub fn run(&self, tls: VMWorkerThread) {
+        // Initialize the GC worker for coordinator. We are not using the run() method from
+        // GCWorker so we manually initialize the worker here.
+        self.scheduler
+            .read()
+            .unwrap()
+            .as_ref()
+            .unwrap()
+            .initialize_coordinator_worker(tls);
+
```

Comment:
The `as_ref().unwrap()` part is for getting into the `Option<_>` in `RwLock<Option<Arc<GCWorkScheduler<VM>>>>`.

This long sequence may be a consequence of MMTk using two-phase initialisation.  It is considered by Rust as an anti-pattern, but considering that GC is very low-level, there may be reasons why it has to be done this way.  Let's investigate it later.

Reply:
We have an issue tracking this: https://github.com/mmtk/mmtk-core/issues/57. In this particular case, I don't think it is because of GC being low-level.